taskcluster/client.py

```diff
@@ -536,9 +536,19 @@ def _makeHttpRequest(self, method, route, payload):
 
             # Try to load JSON
             try:
+                if response.headers or headers in response:
```

Comment:
This has a bit of a bug in it: `headers in response` is likely not what you want.  If there's a `headers` variable in scope here, it's almost certainly not what you're wanting to check for presence in the response object.

The Python 'repl' is really helpful in situations like this for figuring out what's going on here.  Since we know that this response object is what the request library returns from a request, we can get a sample response to do some checks against, like this:

```python
$ python
Python 2.7.14 (default, Mar 14 2018, 13:36:31) 
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> response = requests.get('http://www.google.com');
>>> response.headers
{'Content-Length': '4890', 'X-XSS-Protection': '1; mode=block', 'Content-Encoding': 'gzip', 'Set-Cookie': '1P_JAR=2018-04-10-09; expires=Thu, 10-May-2018 09:37:53 GMT; path=/; domain=.google.de, NID=127=r3tD74HByNeU2WwhWVw6T6AzkeHfem__la1lGpU_g5v3db14PheOHM0tlCMl-t_JgVR3Q7-j5TX21TiN-G-WW0wnwKog1b-jWpDi3b2lqLkuGY--Hvik1dBdkyOn4Oxs; expires=Wed, 10-Oct-2018 09:37:53 GMT; path=/; domain=.google.de; HttpOnly', 'Expires': '-1', 'Server': 'gws', 'Cache-Control': 'private, max-age=0', 'Date': 'Tue, 10 Apr 2018 09:37:53 GMT', 'P3P': 'CP="This is not a P3P policy! See g.co/p3phelp for more info."', 'Content-Type': 'text/html; charset=ISO-8859-1', 'X-Frame-Options': 'SAMEORIGIN'}
>>> headers in response
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'headers' is not defined
>>> 'Content-Type' in response.headers
True
>>> 'Content-Length' in response.headers
True
>>> 
```

Reply:
ah thats true, sorry I didn't pay attention to this. I fixed it